couchbase_test.go

```diff
@@ -0,0 +1,758 @@
+package couchbase
+
+import (
+	"context"
+	"fmt"
+	docker "github.com/hashicorp/vault/helper/testhelpers/docker"
+	"github.com/hashicorp/vault/sdk/database/dbplugin"
+	"github.com/ory/dockertest"
+	dc "github.com/ory/dockertest/docker"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+)
+
+var containerInitialized bool = false
```

Comment:
With the new structure of the tests, we can get rid of this mutable global state. `containerInitialized`, `cleanup`, and `pre6dot5` can all be easily removed or made local. The rest can be made `const` in a block like so:

```go
const (
    adminUsername = "Administrator"
    adminPassword = "password"
    bucketName = "travel-sample"
)
```

Reply:
If you don't mind, I'd prefer to leave pre6dot5 as a global mutable variable, otherwise I am going to either have to call the check 12 more times or pass it through a boat load of function calls.
Others are gone.